lib/rubyXL/convenience_methods.rb

```diff
@@ -856,15 +855,33 @@ def change_text_wrap(wrap = false)
       self.style_index = workbook.modify_alignment(self.style_index) { |a| a.wrap_text = wrap }
     end
 
-    def change_border(direction, weight)
+    def change_border(direction, weight, direct_match = false)
       validate_worksheet
-      self.style_index = workbook.modify_border(self.style_index, direction, weight)
+      merged_cell = !direct_match && worksheet.merged_cells && worksheet.merged_cells.detect { |mc| mc.cover?(row, column) }
+      if merged_cell
+        merged_cell.ref.row_range.each do |r|
+          merged_cell.ref.col_range.each do |c|
+            worksheet.add_cell(r, c, '', nil, false).change_border(direction, weight, :direct_match)
```

Comment:
The problem here is that if merged cell is, like, 3 cells wide and 3 cells high, then  the central cell (which technically does not have any visible borders) will get a border as well...

Also, what will happen if a cell already exists at those coordinates?